src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -177,12 +184,43 @@ async def _run_impl(self) -> None:
                 else:
                     self._buffer.update(sample)
 
+                self.count_samples += 1
+                if self.wait_for_num_samples != 0:
+                    if self.count_samples % self.wait_for_num_samples == 0:
+                        self.wait_for_samples_event.set()
+
         except asyncio.CancelledError:
             _logger.info("MovingWindow task has been cancelled.")
             raise
 
         _logger.error("Channel has been closed")
 
+    async def wait_for_samples(self, num_samples: int) -> None:
+        """Wait until a given number of samples has been received.
+
+        The sample counter is updated irrespective of whether this
+        method is called or not. Thus this might trigger when
+        a smaller number of samples than the given number has been
+        updated.
+
+        Args:
+            num_samples: The number of samples to wait for.
+
+        Raises:
+            ValueError: if the number of samples is less than or equal to zero.
+        """
+        if num_samples <= 0:
+            raise ValueError(
+                "The number of samples to wait for should be greater than zero."
+            )
+
+        if self.count_samples % num_samples == 0:
+            return
+
+        self.wait_for_num_samples = num_samples
+        await self.wait_for_samples_event.wait()
```

Comment:
If you never call `self.wait_for_samples_event.clear()` it will always fire immediately in `.wait()`